### ISSUE TITLE
Freeze selections

### ISSUE THREAD

**Issue by tweakimp** [RationalCoding/multihack-brackets#20]
It would be helpful if you could highlight code for othere, maybe via right click on selected code (right click opens context menu > choose "highlight in multihack." or highlight a whole row via right click on its number.

**Comment by t-mullen** [comment]
Selecting text will highlight it for others, is that what you mean?

**Comment by tweakimp** [comment]
No. If I have text selected, I want to have the option of highlighting it for others.
Otherwise text would be highlighted all over the place while just working on the code. All room mates would also need to have the option to remove all highlights to prevent the code from looking too cluttered.

**Comment by t-mullen** [comment]
Oh you mean leaving a permanent highlight so that when you deselect it, it's still highlighted.

Ok, I'll think about it.

**Comment by tweakimp** [comment]
Exactly. Additionally even give the option to comment on that highlight

**Comment by Worie** [comment]
Hi, this feature is sounds cool, feel free to mention me if you'd like any help with it, I've already created something similar for my private extensions

**Comment by tweakimp** [comment]
This is github, why don't you just start and make a pull request? :)

**Comment by t-mullen** [comment]
@Worie I'd appreciate it. 

I know how to make selections visible, but the issue is persisting the marked text when the user switches editors. Brackets doesn't store any metadata for that. I thought about storing it myself, but never got around to it.

If you have a better solution, please share! :)

**Comment by tweakimp** [comment]
From my newbee view: cant you just assign one person in the group to be the  "master" which means he stores this data? If he leaves the room, someone else becomes master and loads the data from the old master.

**Comment by t-mullen** [comment]
I've figured out how to store the data, so that's alright.

What would the workflow for this be?

I was thinking:
1. User highlights the code.
2. User right-clicks, "Freeze Highlight"
3. Highlight stays there until user right-clicks it, "Remove Highlight"

**Comment by tweakimp** [comment]
I think: right click - > "highlight for all room mates", maybe even "highlight for specific room mate". Everyone who can see the highlight can remove it.

**Comment by t-mullen** [comment]
But you can already see each other's highlights right? All this feature is adding is the ability to freeze a selection.

I like the idea of removing other's highlights though.

**Comment by tweakimp** [comment]
Yeah, I can see the current selection, but this should be a permanent code marker. (until removed)

**Comment by t-mullen** [comment]
I'm still not clear on why you want this. What's the usecase for this that normal selections don't cover?

**Comment by tweakimp** [comment]
Lets say we both code something together.
I look over what you are doing and spot some lines of code that could be done better. I highlight them and tell you "Look over here, redo this". I dont have to keep it selected and can continue working on whatever i am doing, but the selected lines of code are still there and help you find what I meant.

**Comment by t-mullen** [comment]
I think that's better solved by "Bring to line".

[https://github.com/RationalCoding/multihack-web/issues/27](https://github.com/RationalCoding/multihack-web/issues/27)

**Comment by tweakimp** [comment]
I thought of it more like a "to do list" feature, to be able to highlight everything that needs rework.

**Comment by t-mullen** [comment]
Ok, I see why that could be useful. Going to backlog this because it's not core-featureset; but I can see it being implemented later.

**Comment by tweakimp** [comment]
Ok, thanks.
